fili-core/src/main/java/com/yahoo/bard/webservice/data/cache/DataCache.java

```diff
@@ -33,6 +33,27 @@
      */
     boolean set(String key, T value) throws IllegalStateException;
 
+    /**
+     * Put a value on a key in a data cache with the specified expiration.
+     *
+     * By default, this method throws away the expiration.
+     *
+     *
+     * @param key  the key under which this object should be added.
+     * @param value  the object to store
+     * @param expiration The expiration time, it may either be Unix time (number of seconds since January 1, 1970, as
+     * a 32-bit value), or a number of seconds starting from current time. In the latter case, this number of seconds
+     * may not exceed 60*60*24*30 (number of seconds in 30 days); if the number sent by a client is larger than that,
+     * the server will consider it to be real Unix time value rather than an offset from current time.
```

Comment:
Shouldn't the system have an opinion on whether this is a seconds remaining or a time of expiration?

If we need to support the expire-at-time use case perhaps we should have it take a Date instead for specificity.

Reply:
Yeah, I wasn't sure how much meaning we wanted to impose at the interface level, so I basically took what we have in memcache and threw it here as a starting point. I'm ok with making it a Date.